app/src/main/java/io/intrepid/contest/BaseSlideFragment.java

```diff
@@ -0,0 +1,31 @@
+package io.intrepid.contest;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
+import io.intrepid.contest.base.BaseFragment;
+import timber.log.Timber;
+
+public abstract class BaseSlideFragment extends Fragment {
+
+    public abstract boolean canMoveFurther();
+    public abstract int cantMoveFurtherErrorMessage();
```

Comment:
It's not very clear that this is a resource id, maybe you can return a String and inside the method get the actual value with getResources.getString(R.id._____)